ppx/tools/ppx_deriving_tools.ml

```diff
@@ -162,6 +173,30 @@ module Schema = struct
                Location.raise_errorf ~loc
                  "this cannot be a type parameter"))
 
+  let derive_sig_type_decl ~derive_t ~derive_label ~ctxt (_rec_flag, tds)
+      =
+    let loc = Expansion_context.Deriver.derived_item_loc ctxt in
+    List.map tds ~f:(fun td ->
+        let name = td.ptype_name in
+        let type_ = derive_t ~loc name (gen_type_ascription td) in
+        let type_ =
+          List.fold_left (List.rev td.ptype_params) ~init:type_
+            ~f:(fun acc (t, _) ->
+              let loc = t.ptyp_loc in
+              let name =
+                match t.ptyp_desc with
+                | Ptyp_var txt -> { txt; loc }
+                | _ ->
```

Comment:
Why do `derive_of_type_declaration` and `gen_type_ascription` support `Ptyp_any` but not here?

Edit: oh, because we need a name to keep drilling I guess 🤔